---
title: "DESeq2"
output: html_document
---

```{r setup, include=FALSE}
require("knitr")
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "C:\\GitHub\\Lowry-Hexadecenoic\\deseq_outputs\\")
library(dplyr)
library(DESeq2)
library(ggplot2)
library(magrittr)
```

#Load counts and sample info
```{r}
counts = read.table("counts.txt", header=TRUE, row.names=1, sep="\t")
sortedCounts = read.table("sortedCounts.txt", header=TRUE, row.names=1, sep="\t")
coldata = read.table("coldata.txt", header=TRUE, sep="\t")
```

#remove length from counts table
```{r}
counts['Length'] = NULL
sortedCounts['Length'] = NULL
```

# Filter data to show different cohorts

```{r}
require(dplyr)
counts1hr <- select(counts, contains("1h"))
sortedCounts1hr <- select(sortedCounts, contains("1h"))
coldata1hr <- filter(coldata, time=="1hr")
coldata1hr['time'] = NULL

counts6hr <- select(counts, contains("6h"))
sortedCounts6hr <- select(sortedCounts, contains("6h"))
coldata6hr <- filter(coldata, time=="6hr")
coldata6hr['time'] = NULL
```

#sort coldata so it matches counts table columns
```{r}
# coldata_sort <-coldata[order(coldata$sample),]
```


#make deseq2 object
```{r}
dds = DESeqDataSetFromMatrix(countData = counts6hr, colData = coldata6hr, design = ~ mouse + treatment)
```

#transform counts for PCA plots
```{r}
rld <- rlog(dds, blind=FALSE)
vst <- vst(dds, blind=FALSE)
ntd <- normTransform(dds)
```

#plot SD of transformed data
```{r}
setwd("C:\\GitHub\\Lowry-Hexadecenoic\\deseq_outputs\\pdfs\\")
library("vsn")
pdf("sdplot_6hr_shiftedLogtransformed.pdf")
meanSdPlot(assay(ntd))
dev.off()

pdf("sdplot_6hr_regLogtransformed.pdf")
meanSdPlot(assay(rld))
dev.off()

pdf("sdplot_6hr_varstabilizedtransformed.pdf")
meanSdPlot(assay(vst))
dev.off()
```

#check to see what is the dominant driver of gene expression differences
```{r}
setwd("C:\\GitHub\\Lowry-Hexadecenoic\\deseq_outputs\\pdfs\\")

pdf("6hr_treatment_PCA.pdf")
sampleDists <- dist(t(assay(ntd)))
plotPCA(rld, intgroup=c("treatment"))
dev.off()

pdf("time_PCA.pdf")
sampleDists <- dist(t(assay(ntd)))
plotPCA(rld, intgroup=c("time"))
dev.off()
```


# 6hr Treatment Group -----------------------------------------------------------------------------------------------------------------------------------

# Effect of 10(Z)-HA administration for 6 hours:
```{r}
#Designate File Output
setwd("C:\\GitHub\\Lowry-Hexadecenoic\\deseq_outputs\\s6hr\\")

dds = DESeqDataSetFromMatrix(countData = sortedCounts6hr, colData = coldata6hr, design = ~  mouse + treatment)

#remove low count genes (anything less than 20 reads)
# keep <- rowSums(counts(dds)) >= 20
# dds <- dds[keep,]

#set comparison level, 
dds$treatment <- relevel(dds$treatment, ref = "untreated")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("treatment_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="6hr_normalized_counts.csv")

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="6hr_treatment_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("6hr_treatment_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```

# 1hr Treatment Group -----------------------------------------------------------------------------------------------------------------------------------

# Effect of 10(Z)-HA administration for 1 hour:
```{r}
#Designate File Output
setwd("C:\\GitHub\\Lowry-Hexadecenoic\\deseq_outputs\\1hr\\")

dds = DESeqDataSetFromMatrix(countData = counts1hr, colData = coldata1hr, design = ~ treatment)

#remove low count genes (anything less than 20 reads)
keep <- rowSums(counts(dds)) >= 20
dds <- dds[keep,]

#set comparison level, 
dds$treatment <- relevel(dds$treatment, ref = "untreated")

#perform normalization, dispersion estimate, and statistical tests
dds = DESeq(dds)

#plot dispersions
pdf("treatment_dispersionPlot.pdf")
plotDispEsts(dds )
dev.off()

#write out normalized counts
normed = counts(dds, normalized=TRUE)
write.csv(normed, file="1hr_normalized_counts.csv")

#write out statistical results
res <- results(dds)
write.csv(as.data.frame(res), file="1hr_treatment_results.csv")

#plot fold changes with 2 fold blue line added, alpha is FDR cutoff, change to whatever cutoff you want
pdf("1hr_treatment_MAPlot.pdf")
DESeq2::plotMA(res, alpha = 0.1, ylim=c(-3,3), cex=.4)
abline(h=c(-1,1), col="dodgerblue", lwd=2)
dev.off()

```
